controls/overlibmws/overlibConfig.txt

Below are described the configuration variables and arrays associated with the
overlib commands, and their defaults as set in the
	overlibmws.js
core module and the
	overlibmws_bubble.js, overlibmws_crossframe.js, overlibmws_debug.js,
 	overlibmws_draggable.js, overlibmws_exclusive.js, overlib_filter.js,
 	overlibmws_function.js, overlibmws_hide.js, overlibmws_scroll.js and
        overlibmws_shadow.js
plugin modules.  The
	overlibmws_iframe.js, overlibmws_overtwo.js, and overlib_regCore.js
plugin modules do not have configuration variables or arrays.

You can change any of the configuration default values for all pages by making
the modifications in overlibmws.js or the plugin modules, or for individual HTML
pages by declaring any of these variables with the values you prefer in a SCRIPT
block or imported js file for those pages.

You instead can change the configuration default values for individual HTML pages by
calling the OLpageDefaults(arguments) function in a SCRIPT block or imported js file
with "arguments" consisting of a comma-separated list of uppercase command names and
their parameters if any, homologously to the arguments for an overlib() call.

The overlibmws_overtwo.js module has no additional commands, and thus no configuration
variables. This module is used by calling overlib2() and nd2() from within an overlib()
call that invokes a primary popup, to invoke and close secondary popups.

The overlibmws_iframe.js module has no additional commands, and thus no configuration
variables. It should be imported when a page has system controls (e.g., some form
elements, flash objects, applets) which obscure overlib popups.  It corrects this
problem for IE v5.5 or higher.  For versions of IE lower than v5.5 and for other browsers,
you can use commands in the overlibmws_hide.js plugin module.  See the examples in
http://www.macridesweb.com/oltest/hide.html and http://www.macridesweb.com/oltest/flash.html
on how to use those command to hide the system controls when overlib popups are invoked.

The overlibmws_regCore.js module has no overlib() or overlib2() commands, and thus
no configuration variables.  It should be imported in frames which will not themselves
import the core module and any plugin modules, but instead will use those in another
frame.  See its header for more information.  Examples of its use are in
http://www.macridesweb.com/oltest/testFrame.html

See the overlibmws Command Reference (http://www.macridesweb.com/oltest/commandRef.html)
for more information about the commands, configuration variables, and plugin modules. 


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR CORE MODULE overlibmws.js
////////////////////////////////////////////////////////////////////////////////////

// FGCOLOR - Main background color (the large area).
// Usually a bright color (white, yellow etc).
var ol_fgcolor = "#CCCCFF";

// BGCOLOR - Border color.
// Usually a dark color (black, brown etc).
var ol_bgcolor = "#333399";

// CGCOLOR - Caption background color (typically same as border color).
// Usually a dark color (black, brown etc).
var ol_cgcolor = "#333399";

// TEXTCOLOR - Text color.
// Usually a dark color.
var ol_textcolor = "#000000";

// CAPCOLOR - Color of the caption text.
// Usually a bright color
var ol_capcolor = "#FFFFFF";

// CLOSECOLOR - Color of "Close" when using Sticky.
// Usually a semi-bright color.
var ol_closecolor = "#9999FF";

// TEXTFONT - Font face for the main text.
var ol_textfont = "Verdana,Arial,Helvetica";

// CAPTIONFONT - Font face for the caption.
var ol_captionfont = "Verdana,Arial,Helvetica";

// CLOSEFONT - Font face for the close text.
var ol_closefont = "Verdana,Arial,Helvetica";

// TEXTSIZE - Font size for the main text.
var ol_textsize = "1";

// CAPTIONSIZE - Font size for the caption.
var ol_captionsize = "1";

// CLOSESIZE - Font size for the close text.
var ol_closesize = "1";

// FGCLASS - Main background class.
var ol_fgclass = "";

// BGCLASS - Frame background class.
var ol_bgclass = "";

// CGCLASS - Caption background class.
var ol_cgclass = "";

// TEXTPADDING - Padding for main text.
var ol_textpadding = "2";

// TEXTFONTCLASS - Main font class.
var ol_textfontclass = "";

// CAPTIONPADDING - Padding for caption (including Close text if present).
var ol_captionpadding = "2";

// CAPTIONFONTCLASS - Caption font class.
var ol_captionfontclass = "";

// CLOSEFONTCLASS - Close font class.
var ol_closefontclass = "";

// CLOSECLICK - If the user has to click to close stickies.
var ol_closeclick = 0;

// CLOSETEXT - Text for the closing sticky popups. Normal is "Close".
var ol_close = "Close";

// CLOSETITLE - Text to use as value of TITLE attribute for browser-generated TooTips
// with the "Close" link in captions of stickies when CLOSECLICK is changed to 1;
var ol_closetitle = "Click to Close";

// Default text for popups
// Should you forget to pass something to overLIB this will be displayed.
var ol_text = "Default Text";

// Default caption
// You should leave this blank or you will have problems making non caps popups.
var ol_cap = "";

// CAPBELOW - Whether the caption should appear below the main text area.  Default is
// off (0) such that the caption appears above.
var ol_capbelow=0;

// BACKGROUND - Default background image. Better left empty unless you always want one.
var ol_background = "";

// WIDTH - Default width of the popups in pixels. 100-300 pixels is typical.
// This value is simply a suggestion to the browser, which may change the
// actual width depending on the content.
var ol_width = "200";

// WRAP - Intended to keep the popup no wider than its content plus normal padding, but
// to wrap the content if it would exceed the window width, or if it would exceed WRAPMAX
// when that has been set to a value greater than zero.
// Overrides the o3_width setting. Default is no wrap (0).
var ol_wrap = 0;

// WRAPMAX - If set to a value greater than 0, sets the maximum width of the popup, up to
// the window width, before wrapping occurs when the WRAP command is set.
var ol_wrapmax = 0;

// HEIGHT - Default height for popup. Often best left alone.
var ol_height = -1;

// BORDER - How thick the ol_border should be in pixels.
// 1-3 pixels is typical.
var ol_border = "1";

// BASE - Any additional thickening of the border's base in pixels.
var ol_base = "0";

// OFFSETX - How many pixels to the right (positive values) or left (negative values)
// of the cursor to show the popup. Values between 3 and 12 are best.
var ol_offsetx = 10;
	
// OFFSETY - How many pixels below (positive values) or above (negative values) the
// cursor to show the popup. Values between 3 and 20 are best.
var ol_offsety = 10;

// STICKY - Decides if sticky popups are default. 0 for non, 1 for stickies.
var ol_sticky = 0;

// EXCLUSIVE - Decides if a sticky popup should be exclusive, such that no other popup
// can be invoked and replace it before the sticky is closed by the user or a timeout.
var ol_exclusive = 0;

// NOFOLLOW - Should non-sticky popups not follow cursor movements (i.e., remain
// stationary where initially positioned on invocation, like title-based tooltips).
var ol_nofollow = 0;

// NOCLOSE - Omit Close text in stickies with captions, for all stickies use mouse off
// after mouse over sticky to close, and cancel any timeout while over sticky.
var ol_noclose = 0;

// MOUSEOFF - For stickies which do have a caption with a CLOSETEXT, also use mouse off
// after mouse over sticky to close, and cancel any timeout while over sticky.
var ol_mouseoff = 0;

// OFFDELAY - Default delay for closing NOCLOSE or MOUSEOFF popups.  If a mouse over the
// sticky occurs during this delay, the close is cancelled.
var ol_delay = 300;

// RIGHT - Default vertical alignment for popups.
// It's best to leave RIGHT here. Other options are LEFT and CENTER.
var ol_hpos = RIGHT;

// BELOW - Default vertical position of the popups.
// It's best to leave BELOW here. Other options are ABOVE and VCENTER.
var ol_vpos = BELOW;

// Default status bar text when a popup is invoked.
var ol_status = "";

// AUTOSTATUS, AUTOSTATUSCAP - If the status bar automatically should load either
// text or caption. 0=nothing, 1=text, 2=caption
var ol_autostatus = 0;

// SNAPX - Horizontal grid spacing that popups will snap to.
// 0 makes no grid, anything else will cause a snap to that grid spacing.
var ol_snapx = 0;

// SNAPY - Vertical grid spacing that popups will snap to.
// 0 makes no grid, anything else will cause a snap to that grid spacing.
var ol_snapy = 0;

// FIXX - Sets the popup horizontal position to a fixed column.
// Numbers greater than -1 will cause fixed position.
var ol_fixx = -1;

// FIXY - Sets the popup vertical position to a fixed row.
// Numbers greater than -1 will cause fixed position.
var ol_fixy = -1;

// RELX - Sets the popup horizontal position to a column relative to the window display.
// Anything numeric (non-null) will cause relative position. Positive and 0 is to
// the right from left window margin for left margin of popup. Negative is to the
// left from right window margin for right margin of popup.
var ol_relx = null;

// RELY - Sets the popup vertical position to a row relative to the window display.
// Anything numeric (non-null) will cause relative position. Positive and 0 is down
// from top window margin for top margin of popup. Negaive is up from bottom window
// margin for bottom margin of popup.
if (typeof ol_rely == 'undefined'){var ol_rely = null;}

// MIDX - Sets the popup horizontal midpoint to a column relative to the window horizontal
// midpoint. Anything numeric (non-null) will cause midpoint position. Positive and
// 0 is to the right from the window midpoint. Negative is to the left.
var ol_midx = null;

// MIDY - Sets the popup vertical midpoint to a row relative to the window vertical midpoint.
// Anything numeric (non-null) will cause midpoint position. Positive and 0 is down from
// the window midpoint. Negative is up.
var ol_midy = null;

// REF - The NAME of an anchor or image, or ID of a layer, to serve as a reference object such
// that a corner of the popup will be positioned relative to a corner of the object.
var ol_ref = "";

// REFC - Corner of the reference object for positioning.
// Value can be: 'UL' (Upper Left), 'UR', 'LR', or 'LL'.
var ol_refc = 'UL';

// REFP - Corner of the popup for positioning.
// Value can be: 'UL' (Upper Left), 'UR', 'LR', or 'LL'.
var ol_refp = 'UL';

// REFX - X displacement from the reference point. Positive to the right,
// negative left.
var ol_refx = 0;

// REFY - Y displacement from the reference point. Positive down, negative up.
var ol_refy = 0;

// FGBACKGROUND - Background image for the popup's inside.
var ol_fgbackground = "";

// BGBACKGROUND - Background image for the popup's frame (border).
var ol_bgbackground = "";

// CGBACKGROUND - Background image for the caption.
var ol_cgbackground = "";

// PADX
// How much horizontal left padding text should get by default when BACKGROUND is used.
var ol_padxl = 1;
// How much horizontal right padding text should get by default when BACKGROUND is used.
var ol_padxr = 1;

// PADY
// How much vertical top padding text should get by default when BACKGROUND is used.
var ol_padyt = 1;
// How much vertical bottom padding text should get by default when BACKGROUND is used.
var ol_padyb = 1;

// FULLHTML - If the user by default must supply all html for complete control of popup content.
// Set to 1 to activate, 0 otherwise.
var ol_fullhtml = 0;

// CAPICON - Default icon to place next to the popups caption.
if (typeof ol_capicon == 'undefined'){var ol_capicon = "";}

// FRAME - Default frame. We default to current frame if there is no frame defined.
var ol_frame = self;

// TIMEOUT - Default timeout. By default there is no timeout.
var ol_timeout = 0;

// DELAY - Default delay for onset of popup. By default there is no delay.
var ol_delay = 0;

// HAUTO - If overLIB should decide the horizontal placement.
var ol_hauto = 0;

// VAUTO - If overLIB should decide the vertical placement.
var ol_vauto = 0;

// NOJUSTX - If overLIB should let popups overrun the left or right window margins.
var ol_nojustx = 0;

// NOJUSTY - If overLIB should let popups overrun the top or bottom window margins.
var ol_nojusty = 0;

////////////////////////////////////////////////////////////////////////////////////
// ARRAY CONFIGURATION FOR CORE MODULE overlibmws.js
////////////////////////////////////////////////////////////////////////////////////

// INARRAY - Array with texts.
var ol_texts = new Array(
 "Text 0",
 "Text 1");

// CAPARRAY - Array with captions.
var ol_caps = new Array(
 "Caption 0",
 "Caption 1");

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_bubble.js
////////////////////////////////////////////////////////////////////////////////////

// BUBBLE - Whether to use a bubble type popup. Default is no (0).
var ol_bubble = 0;

// BUBBLETYPE - Type of bubble image to use. Default is 'flower'.  Other options are
// 'oval', 'square', 'pushpin', 'quotation', or 'roundedcorners'. Specify directory
// for images via the last parameter of registerImages() near the top of the plugin
// module (default value is './').
var ol_bubbletype = 'flower';

// ADJBUBBLE - Whether to resize the image in relation to the content.
// Default is no (0). 
var ol_adjbubble = 0;

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_bubble.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_crossframe.js
////////////////////////////////////////////////////////////////////////////////////

You must import this plugin module to use the FRAME command, but its configuration
variable, ol_frame, is set in the core module.

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_crossframe.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_debug.js
////////////////////////////////////////////////////////////////////////////////////

// ALLOWDEBUG - The ID or a comma-separated list of IDs for debug layers which, if
// invoked following an overlib call and then made hidden via the close link at
// upper right, should be made visible again whenever that overlib call occurs.
var ol_allowdebug = "";

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_debug.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_draggable.js
////////////////////////////////////////////////////////////////////////////////////

// DRAGGABLE - If sticky should be draggable.
var ol_draggable = 0;

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_draggable.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_exclusive.js
////////////////////////////////////////////////////////////////////////////////////

// EXCLUSIVE = If sticky should be exclusive.
var ol_exclusive = 0;

// EXCLUSIVESTATUS - Status line string to use for exclusive stickies
var ol_exclusivestatus = 'Please act on or close the open popup.';

// EXCLUSIVEOVERRIDE = If a displayed exclusive sticky should be overridden by the
// current overlib call for a popup.
var ol_exclusiveoverride=0;

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_exclusive.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_filter.js
////////////////////////////////////////////////////////////////////////////////////

// FILTER - Toggles on the filter feature set for IE v5.5+ browsers. Defualt is off.
var ol_filter=0;

// FADEIN - Filter type for stylinzed fadein. Value can be 0 - 50 for the 51 types,
// or 51 (default) for random selections of the type across successive occurrences
// of the popup.
var ol_fadein=51;

// FADETIME - Duration of fadein (millisec).
var ol_fadetime=800;

// FILTEROPACITY - Opacity of entire popup.  The higher the number in the range of 1-99,
// the more more opaque (less transparent) the popup will be.  But 0 is handled as
// equivalent to 100 (no transparency).  This feature also is implemented for Mozilla
// and Netscape v6+ browsers.
var ol_filteropacity=100;

// FILTERSHADOW - Type of filter-based shadow.  Default is off (0).
// Dropshadow is 1. Shadow (tapers from corners) is 2.
var ol_filtershadow=0;

// FILTERSHADOWCOLOR - Color of filter-based shadow.
var ol_filtershadowcolor="#cccccc";

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_filter.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_function.js
////////////////////////////////////////////////////////////////////////////////////

// FUNCTION - Default javascript function. By default there is none.
var ol_function = null;

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_function.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////
// DEFAULT CONFIGURATION FOR PLUGIN MODULE overlibmws_hide.js
////////////////////////////////////////////////////////////////////////////////////

// HIDESELECTBOXES - Whether to hide any select boxes which overlap the popup
// while the popup is being displayed.
var ol_hideselectboxes=0;

// HIDEBYID - An id or comma-separated list of id's to be hidden while the popup
// is displayed.  Is intended for form elements and is ignored for any browsers
// using HIDESELECTBOXES and for Opera v7+.
var ol_hidebyid='';

// HIDEBYIDALL - An id or comma=separated list of id's to be hidden while the popup
// is displayed.  Is intended for non-form elements with system controls, e.g.,
// flash objects and applets.
var ol_hidebyidall='';

// HIDEBYIDNS4 - An id or comma-separated list of id's for positioned div's to be
// hidden while the popup is being displayed by Netscape v4.x browsers.
var ol_hidebyidns4='';

////////////////////////////////////////////////////////////////////////////////////
// END CONFIGURATION FOR overlibmws_hide.js
////////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////////

CHANGELOG
2007/07/20 Version 2.17-stable-9
-Better import for sync
-Updated Czech language
-Implemented new version of jupload

2007/07/08 Version 2.17-stable-8
-Problem with auto_check in email for new users.

2007/07/05 Version 2.17-stable-7
-Added out of office reply for local e-mail accounts. No cron job is needed 
 anymore but you need to setup sudo.
-Fixed problem in synchronization with strange characters
-Brought back e-mail popup notification
-Automatic e-mail check can be enabled per account
-Updated Czech language
-Updated Slovenian language

2007/06/18 Version 2.17-stable-6
-fixed bug with inline e-mail attachments
-Fixed display of link e-mail button in Community version

2007/06/12 Version 2.17-stable-5
-Better character set conversion in e-mail messages
-Filesystem shares bug with invalid tree structures
-Display inline e-mail attachments
-Other small bugs

2007/06/07 Version 2.17-stable-4
-Added the last 10 uploaded files to the summary
-Calendar and task reminder beeps only once if it's not dismissed.
-E-mail account save bug if you don't have admin permissions
-Custom fields class outputted a blank line which caused an header already sent error

2007/05/31 Version 2.17-stable-3
-E-mail status indicator was missing in all themes except default
-Fixed an issue with strange characters in filenames
-Fixed printing e-mails from a folder with strange characters
-Fixed other small bugs

2007/05/21 Version 2.17-stable-1
-Added Slovenian language
-Updated Turkish language
-Changed e-mail permissions behaviour
-Fixed update procedure error

2007/05/07 Version 2.17-testing-1
-Full support for Microsoft Windows
-Move module theme items to module dir
-Modular select dialog (Done with global search).
-Select and sort columns to display in tables
-global search
-Easier linking with global search
-link email (Pro only)
-Better auto add contacts in mail
-TNEF (Winmail.dat) support
-Reports module to show different queries on addressbook. Can also be used to send
 mailings with MS Office, Open-Office and the GO e-mail client. Replaces old mailing 
 system (Pro only)
-New e-mail counter always in top menu bar and beep can be disabled when new mail 
 arrives
-Option to show attachment paperclips and priorities in e-mail client. Slows it down when
 enabled!
-Don't create a personal addressbook by default for everyone.
-Edit multiple contacts at once
-Show month name on top of calendar print
-Adminstrative tools: check database, backup database, users import
-Help button leads to right page at docs.group-office.com
-Table configuration which allows you to reorder and select the columns you want in tables



CHANGELOG
2007/05/02 Version 2.16-14
-Added update client
-Fixed problem with quotes in e-mail names
-Don't encode language files for translating
-Small bug with newlines and translate module


2007/04/03 Version 2.16-13
-Security issue: A logged in user could read another user's e-mail with some tricks
-Saved e-mails couldn't be read (Pro version)
-Calendar admins always have access to events
-Updated Catalan language


2007/03/08 Version 2.16-12
-Fixed mailto links in e-mail
-Don't set session.save_path because PHP garbage collection doesnt work then 
 anymore. This resulted in a lot of session files in the tmp dir at some distro's
-Autocomplete users only put in a , in the text field. Also an Invalid XML bug was fixed.
-You didn't fill in all required fields bug at registration and profile update

Language updates:
-German language update
-Added Turkish language

2007/02/05 Version 2.16-11
-Fixed some corrupt files

2007/02/05 Version 2.16-10

Bug fixes:
-Files with quotes went wrong in filesystem
-Missing image for adding users
-html chars in pulldown of email module
-Allow duplicate email addresses with differt names in autocomplete
-Sync recurrence 1 day minus bug
-Double file properties bug
-E-mail attachment character encoding
-Email addresses character encoding
-Double frame on user profile close

Languages updated:
-Spanish
-French
-Catalan
-Hungarian
-Italian

2006/11/29 Version 2.16-4
-Translation module
-Speed optimizations



2006/11/01 Version 2.16-FINAL
-Disable user fields in admin mode
-Create pulldown menu's in custom fields. (Professional version)
-Custom fields for users (Professional version)
-Configure welcome e-mail on user registration
-Notify admin on user sign up
-Disable/enable fields for registration
-Book overlapping times in projects module
-Improved installer
-make compatible with register_globals=on

2006/09/13 Version 2.16-FC
-Small bug fixes
-Introduction of Gallery module

2006/08/31 Version 2.16-RC1
-Quotes/Invoices module (Professional version)
-Save e-mail to filesystem and link it to anything (Professional version)
-Improved linking interface
-Add local system users and (Postfix) e-mail aliases with sudo
-Bug fixes

2006/07/12 Version 2.15-FINAL-9
-Save email attachments to a project
-Security fixes
-Background colors in calendar
-One day calendar view in emerged group view
-Events that got accepted were not synced.
-FCKeditor upgraded to 2.3
-Project fees are only visible by name not by the amount.
-Projects print function

2006/06/21 Version 2.15-FINAL-8
-Access denied bug in tasks module.
-Small bug fixes.

2006/05/24 Version 2.15-FINAL-7
-Compatibility release for Quotes/Invoices module.
-Security fix for calendar module
-Websites module can have inline php in templates and can have a template per page.

2006/05/22 Version 2.15-FINAL-6
-Websites module uses FCKeditor
-Websites module can have a different template per file.

2006/05/07 Version 2.15-FINAL-5
-Task sync (Professional version)
-Exceptions in recurring events/tasks get synced (Professional version)
-Only part of the calendar can be synced. For example only appoinments newer then
 30 days old and happen within the next 90 days.
-Company fields of contacts are synced too now.
-E-mail notification on file upload can be enabled/disabled.
-Bug fixes

2006/05/07 Version 2.15-FINAL-4

modules/users/settings.php
<?php
/*
   Copyright Intermesh 2003
   Author: Merijn Schering <mschering@intermesh.nl>
   Version: 1.0 Release date: 08 July 2003

   This program is free software; you can redistribute it and/or modify it
   under the terms of the GNU General Public License as published by the
   Free Software Foundation; either version 2 of the License, or (at your
   option) any later version.
 */

require_once("../../Group-Office.php");
require_once($GO_LANGUAGE->get_language_file('users'));

$GO_SECURITY->authenticate();
$GO_MODULES->authenticate('users');

load_basic_controls();
load_control('htmleditor');

$return_to = 'index.php';

$task = isset($_POST['task']) ? $_POST['task'] : '';



$form = new form('users_settings_form');
$form->add_html_element(new input('hidden', 'task', '', false));
$form->add_html_element(new input('hidden', 'close', '', false));



	
$tabstrip = new tabstrip('users_settings', $cmdSettings,'200');
$tabstrip->set_attribute('style', 'width:100%;height:100%');
$tabstrip->set_return_to($return_to);

$tabstrip->add_tab('disabled_user_fields', $admin_disabled_user_fields);
$tabstrip->add_tab('registration_unconfirmed', $admin_registration_unconfirmed);
$tabstrip->add_tab('registration_confirmation', $admin_registration_confirmation);


switch($tabstrip->get_active_tab_id())
{

	
	case 'registration_confirmation':
		
		if($task == 'save')
		{
			$registration_confirmation = isset($_POST['registration_confirmation']) ? smart_addslashes($_POST['registration_confirmation']) : '';
			$registration_confirmation_subject = isset($_POST['registration_confirmation_subject']) ? smart_addslashes($_POST['registration_confirmation_subject']) : '';

			
			$GO_CONFIG->save_setting('registration_confirmation', $registration_confirmation);
			$GO_CONFIG->save_setting('registration_confirmation_subject', $registration_confirmation_subject);
			if($_POST['close']=='true')
			{
				header('Location: '.$return_to);
				exit();
			}
		}
		
		$registration_confirmation = $GO_CONFIG->get_setting('registration_confirmation');
		
		if(!$registration_confirmation)
		{
			$registration_confirmation='';
		}
		
		$registration_confirmation_subject = $GO_CONFIG->get_setting('registration_confirmation_subject');
		
		if(!$registration_confirmation_subject)
		{
			$registration_confirmation_subject='';
		}
		
		
		$table = new table();
		$table->set_attribute('style','width:100%;height:100%');
		
		$cell = new table_cell($admin_registration_confirmation_text);
		$row = new table_row();
		$row->add_cell($cell);
		$table->add_row($row);
		
		if(isset($feedback))
		{
			$cell = new table_cell($feedback);
			$cell->set_attribute('class','Error');
			$row = new table_row();
			$row->add_cell($cell);
			$table->add_row($row);
		}				
		
		$subtable = new table();
		$row = new table_row();
		$row->add_cell(new table_cell($admin_subject.': '));
		$input = new input('text', 'registration_confirmation_subject',$registration_confirmation_subject);
		$input->set_attribute('maxlength','100');
		$row->add_cell(new table_cell($input->get_html()));
		$subtable->add_row($row);
			
		$row = new table_row();
		$row->add_cell(new table_cell($subtable->get_html()));
		$table->add_row($row);
		
		
		$htmleditor = new htmleditor('registration_confirmation');
		$htmleditor->setAutoDataDefinitionsPath($GO_MODULES->modules['users']['path'].'auto_data_fields.inc');
		
		$htmleditor->Value=$registration_confirmation;
		
		$htmleditor->SetConfig('CustomConfigurationsPath', $GO_MODULES->modules['users']['url'].'fckconfig.js');
		$htmleditor->ToolbarSet='Users';
		$cell = new table_cell($htmleditor->CreateHtml());
		
		$cell->set_attribute('style', 'height:100%');		
		$row = new table_row();
		$row->add_cell($cell);
		$table->add_row($row);
		
		$row = new table_row();
		$cell = new table_cell();
		$cell->add_html_element(new button($cmdOk, 'javascript:save(\'save\', \'true\');'));
		$cell->add_html_element(new button($cmdApply, 'javascript:save(\'save\', \'false\');'));
		$cell->add_html_element(new button($cmdCancel, 'javascript:document.location=\''.$return_to.'\';'));
		$row->add_cell($cell);
		$table->add_row($row);
		
		$tabstrip->add_html_element($table);
		break;
		
	case 'registration_unconfirmed':
		if($task == 'save')
		{
			$registration_unconfirmed = isset($_POST['registration_unconfirmed']) ? smart_addslashes($_POST['registration_unconfirmed']) : '';
			$registration_unconfirmed_subject = isset($_POST['registration_unconfirmed_subject']) ? smart_addslashes($_POST['registration_unconfirmed_subject']) : '';

			
			$GO_CONFIG->save_setting('registration_unconfirmed', $registration_unconfirmed);
			$GO_CONFIG->save_setting('registration_unconfirmed_subject', $registration_unconfirmed_subject);
			if($_POST['close']=='true')
			{
				header('Location: '.$return_to);
				exit();
			}
		}
		
		$registration_unconfirmed = $GO_CONFIG->get_setting('registration_unconfirmed');
		
		if(!$registration_unconfirmed)
		{
			$registration_unconfirmed='';
		}
		
		$registration_unconfirmed_subject = $GO_CONFIG->get_setting('registration_unconfirmed_subject');
		
		if(!$registration_unconfirmed_subject)
		{
			$registration_unconfirmed_subject='';
		}
		
		
		$table = new table();
		$table->set_attribute('style','width:100%;height:100%');
		
		$cell = new table_cell($admin_registration_unconfirmed_text);
		$row = new table_row();
		$row->add_cell($cell);
		$table->add_row($row);
		
		if(isset($feedback))
		{
			$cell = new table_cell($feedback);
			$cell->set_attribute('class','Error');
			$row = new table_row();
			$row->add_cell($cell);
			$table->add_row($row);
		}				
		
		$subtable = new table();
		$row = new table_row();
		$row->add_cell(new table_cell($admin_subject.': '));
		$input = new input('text', 'registration_unconfirmed_subject',$registration_unconfirmed_subject);
		$input->set_attribute('maxlength','100');
		$row->add_cell(new table_cell($input->get_html()));
		$subtable->add_row($row);
			
		$row = new table_row();
		$row->add_cell(new table_cell($subtable->get_html()));
		$table->add_row($row);
		
		
		$htmleditor = new htmleditor('registration_unconfirmed');
		$htmleditor->setAutoDataDefinitionsPath($GO_MODULES->modules['users']['path'].'auto_data_fields.inc');
		
		$htmleditor->Value=$registration_unconfirmed;
		
		$htmleditor->SetConfig('CustomConfigurationsPath', $GO_MODULES->modules['users']['url'].'fckconfig.js');
		$htmleditor->ToolbarSet='Users';
		$cell = new table_cell($htmleditor->CreateHtml());
		
		$cell->set_attribute('style', 'height:100%');		
		$row = new table_row();
		$row->add_cell($cell);
		$table->add_row($row);
		
		$row = new table_row();
		$cell = new table_cell();
		$cell->add_html_element(new button($cmdOk, 'javascript:save(\'save\', \'true\');'));
		$cell->add_html_element(new button($cmdApply, 'javascript:save(\'save\', \'false\');'));
		$cell->add_html_element(new button($cmdCancel, 'javascript:document.location=\''.$return_to.'\';'));
		$row->add_cell($cell);
		$table->add_row($row);
		
		$tabstrip->add_html_element($table);
		break;
		
	
	default:

	
		
		if($task == 'save')
		{
			$fields = isset($_POST['fields']) ? $_POST['fields'] : array();
			$disabled_user_fields = implode(',', $fields);
			
			$GO_CONFIG->save_setting('disabled_user_fields', $disabled_user_fields);
			if($_POST['close']=='true')
			{
				header('Location: '.$return_to);
				exit();
			}
		}
		
		$disabled_user_fields = $GO_CONFIG->get_setting('disabled_user_fields');
		
		if(!$disabled_user_fields)
		{
			$disabled_user_fields=array();
		}else {
			$disabled_user_fields = explode(',', $disabled_user_fields);
		}
		
			$fields = array(
			'first_name'=>$strFirstName,
			'middle_name'=>$strMiddleName,
			'last_name'=>$strLastName,
			'initials'=>$strInitials,
			'title'=>$strTitle,
			'birthday'=>$strBirthday,
			'email'=>$strEmail,
			'home_phone'=>$strPhone,
			'work_phone'=>$strWorkphone,
			'fax'=>$strFax,
			'cellular'=>$strCellular,
			'country_id'=>$strCountry,
			'state'=>$strState,
			'city'=>$strCity,
			'zip'=>$strZip,
			'address'=>$strAddress,
			'address_no'=>$strAddressNo,
			'company'=>$strCompany,
			'department'=>$strDepartment,
			'function'=>$strFunction,
			'work_country_id'=>$strWorkCountry,
			'work_state'=>$strWorkState,
			'work_city'=>$strWorkCity,
			'work_zip'=>$strWorkZip,
			'work_address'=>$strWorkAddress,
			'work_address_no'=>$strWorkAddressNo,
			'work_fax'=>$strWorkFax,
			'homepage'=>$strHomepage,
			'sex'=>$strSex
			);
			
		if(isset($GO_MODULES->modules['custom_fields']))
		{
			require_once($GO_MODULES->modules['custom_fields']['class_path'].'custom_fields.class.inc');
			$cf = new custom_fields();
			$cf2 = new custom_fields();
			
			$cf->get_categories(8);
			
			while($cf->next_record())
			{
				$cf2->get_fields($cf->f('id'));
				while($cf2->next_record())
				{
					$fields['col_'.$cf2->f('id')]=$cf2->f('name');				
				}
			}			
		}
			
		$tabstrip->add_html_element(new html_element('p', $disabled_user_fields_text));
		
		$table = new table();
		
		$row = new table_row();
		
		foreach($fields as $field=>$name)
		{
			$checkbox = new checkbox($field, 'fields[]',$field, $name, in_array($field, $disabled_user_fields));
			$row->add_cell(new table_cell($checkbox->get_html()));
		
			if(count($row->cells)==2)
			{
				$table->add_row($row);
				$row = new table_row();	
			}	
		}
		if(count($row->cells)==1)
		{
			$table->add_row($row);
		}
		
		$tabstrip->add_html_element($table);
		
		$tabstrip->add_html_element(new button($cmdOk, 'javascript:save(\'save\', \'true\');'));
		$tabstrip->add_html_element(new button($cmdApply, 'javascript:save(\'save\', \'false\');'));
		$tabstrip->add_html_element(new button($cmdCancel, 'javascript:document.location=\''.$return_to.'\';'));
	break;
}





$form->add_html_element($tabstrip);
require_once($GO_THEME->theme_path."header.inc");
echo $form->get_html();
?>
<script type="text/javascript">
function save(task, close)
{
	document.users_settings_form.task.value=task;
	document.users_settings_form.close.value=close;
	document.users_settings_form.submit();
}
</script>
<?php
require_once($GO_THEME->theme_path."footer.inc");
